app/Http/Requests/BulkUserUpdateRequest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Models\User;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class BulkUserUpdateRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $user_id = $this->input('users.*.id');
+        $circle_id = $this->route('circle_id');
+        return [
+            'users' => ['required', 'array', 'min:1'],
+            'users.*.id' => ['required', 'integer',
+                // validate id exists as a user in circle
+                Rule::exists('users')->where(function ($query) use ($circle_id) {
+                    return $query->where('circle_id', $circle_id)->whereNull('deleted_at');
+                })],
+            'users.*.address' => ['nullable', 'string', 'size:42', 'distinct',
+                // validate address is unique within circle
+                Rule::unique('users')->where(function ($query) use ($circle_id, $user_id) {
+                    return $query->where('circle_id', $circle_id)->whereNotIn('id', $user_id)->whereNull('deleted_at');
+                })],
+            'users.*.name' => ['nullable', 'string', 'max:255'],
```

Comment:
noticing this is `nullable` during Update, but `required` during Create. Am I understanding that right?

Reply:
changed to required, but should never trigger since if not provided will be refilled with existing data